packages/bpk-component-select/src/BpkSelect.js

```diff
@@ -15,81 +15,124 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+/* @flow */
 
-import PropTypes from 'prop-types';
+import PropTypes, { type Node } from 'prop-types';
 import React from 'react';
 import { cssModules } from 'bpk-react-utils';
 
 import STYLES from './bpk-select.scss';
 
 const getClassName = cssModules(STYLES);
 
-const BpkSelect = props => {
-  const classNames = [getClassName('bpk-select')];
+export type Props = {
+  id: string,
+  name: string,
+  value: string,
+  className: ?string,
+  docked: boolean,
+  dockedFirst: boolean,
+  dockedLast: boolean,
+  dockedMiddle: boolean,
+  image: ?Node,
+  imageContainerClassName: ?string,
+  large: boolean,
+  valid: ?boolean,
+  wrapperClassName: ?string,
+};
+
+const BpkSelect = (props: Props) => {
   const {
-    valid,
-    large,
+    className,
     docked,
     dockedFirst,
-    dockedMiddle,
     dockedLast,
-    className,
+    dockedMiddle,
+    image,
+    imageContainerClassName,
+    large,
+    valid,
+    wrapperClassName,
     ...rest
   } = props;
 
   // Explicit check for false primitive value as undefined is
   // treated as neither valid nor invalid
   const isInvalid = valid === false;
 
-  if (large) {
-    classNames.push(getClassName('bpk-select--large'));
-  }
-  if (docked) {
-    classNames.push(getClassName('bpk-select--docked'));
-  }
-  if (dockedFirst) {
-    classNames.push(getClassName('bpk-select--docked-first'));
-  }
-  if (dockedMiddle) {
-    classNames.push(getClassName('bpk-select--docked-middle'));
-  }
-  if (dockedLast) {
-    classNames.push(getClassName('bpk-select--docked-last'));
-  }
-  if (className) {
-    classNames.push(className);
-  }
-
-  return (
+  const select = (
     <select
-      className={classNames.join(' ')}
+      className={getClassName(
+        'bpk-select',
+        large && 'bpk-select--large',
+        docked && 'bpk-select--docked',
+        dockedFirst && 'bpk-select--docked-first',
+        dockedMiddle && 'bpk-select--docked-middle',
+        dockedLast && 'bpk-select--docked-last',
+        className,
+      )}
       aria-invalid={isInvalid}
       {...rest}
     />
   );
+
+  if (image) {
+    return (
+      <div
+        className={getClassName(
+          'bpk-select',
+          'bpk-select--with-image',
+          large && 'bpk-select--large',
+          docked && 'bpk-select--docked',
+          dockedFirst && 'bpk-select--docked-first',
+          dockedMiddle && 'bpk-select--docked-middle',
+          dockedLast && 'bpk-select--docked-last',
+          wrapperClassName,
```

Comment:
This isn't valid BEM, as the class names are being re-used on this and the select element - so I think we need to introduce some repetition in the Sass to make these unique.